src/Scripting/Core/Hosting/CommandLine/CommandLineRunner.cs

```diff
@@ -109,14 +109,17 @@ private int RunInteractiveCore(ErrorLogger errorLogger)
             }
 
             var scriptPathOpt = sourceFiles.IsEmpty ? null : sourceFiles[0].Path;
-            var scriptOptions = GetScriptOptions(_compiler.Arguments, scriptPathOpt, _compiler.MessageProvider, diagnosticsInfos);
 
             var errors = _compiler.Arguments.Errors.Concat(diagnosticsInfos.Select(Diagnostic.Create));
             if (_compiler.ReportErrors(errors, _console.Error, errorLogger))
             {
                 return CommonCompiler.Failed;
             }
 
+            // only emit symbols for non-interactive mode, and when encoding is known
+            var emitDebugInformation = !_compiler.Arguments.InteractiveMode && code.Encoding != null;
```

Comment:
correct `var emitDebugInformation = !_compiler.Arguments.InteractiveMode;` would be enough no need to check encoding.
however to emit debug info in interactive, we would need to use some default encoding, since the input doesn't come from a filestream but from string, so at the moment just switching it on causes interactive to fail with "cannot emit debug when encoding is unknown".